cmake/sca/eclair/sca.cmake

```diff
@@ -21,75 +21,117 @@ set(CXX_ALIASES "${CMAKE_CXX_COMPILER}")
 message(STATUS "ECLAIR ECB files have been written to: ${ECLAIR_DATA_DIR}")
 message(STATUS "ECLAIR BUILD DIR is: ${ECLAIR_BUILD_DIR}")
 
 add_custom_target(eclair ALL
-  COMMAND sh -c "ECLAIR_PROJECT_NAME=${ECLAIR_PROJECT_NAME} \
-    ECLAIR_PROJECT_ROOT=${ZEPHYR_BASE} \
-    ECLAIR_DIAGNOSTICS_OUTPUT=${ECLAIR_DIAGNOSTICS_OUTPUT} \
-    ECLAIR_DATA_DIR=${ECLAIR_DATA_DIR} \
-    ECLAIR_RULES_SET=${ECLAIR_RULES_SET} \
-    CC_ALIASES=\"${CC_ALIASES}\" \
-    CXX_ALIASES=\"${CXX_ALIASES}\" \
-    AS_ALIASES=\"${AS_ALIASES}\" \
-    LD_ALIASES=\"${LD_ALIASES}\" \
-    AR_ALIASES=\"${AR_ALIASES}\" \
-    ${ECLAIR_ENV} \
-    -verbose \
-    -eval_file=${ECLAIR_ECL_DIR}/analysis.ecl \
-    -- west build -p always \
-                  -b ${BOARD} \
-                  --build-dir ${ECLAIR_BUILD_DIR} \
-                  ${APPLICATION_SOURCE_DIR} \
-                  | tee ${ECLAIR_OUTPUT_DIR}/analysis.log"
-  BYPRODUCTS ${ECLAIR_OUTPUT_DIR}/analysis.log ${ECLAIR_OUTPUT_DIR}/report.log
-  VERBATIM
-  USES_TERMINAL
-  COMMAND_EXPAND_LISTS
-)
-
-add_custom_command(
-  TARGET eclair POST_BUILD
-  COMMAND sh -c "ECLAIR_DATA_DIR=${ECLAIR_DATA_DIR} \
-  ECLAIR_OUTPUT_DIR=${ECLAIR_OUTPUT_DIR} \
-  ECLAIR_PROJECT_ECD=${ECLAIR_PROJECT_ECD} \
-  ECLAIR_metrics_tab=${ECLAIR_metrics_tab} \
-  ECLAIR_reports_tab=${ECLAIR_reports_tab} \
-  ECLAIR_summary_txt=${ECLAIR_summary_txt} \
-  ECLAIR_summary_doc=${ECLAIR_summary_doc} \
-  ECLAIR_summary_odt=${ECLAIR_summary_odt} \
-  ECLAIR_full_txt_areas=${ECLAIR_full_txt_areas} \
-  ECLAIR_full_txt=${ECLAIR_full_txt} \
-  ECLAIR_full_doc_areas=${ECLAIR_full_doc_areas} \
-  ECLAIR_full_doc=${ECLAIR_full_doc} \
-  ECLAIR_full_odt=${ECLAIR_full_odt} \
-  ${ECLAIR_REPORT} \
-  -eval_file=${ECLAIR_ECL_DIR}/reports.ecl \
-  | tee ${ECLAIR_OUTPUT_DIR}/report.log"
-  COMMAND ${ECLAIR_REPORT} -db=${ECLAIR_PROJECT_ECD} -overall_txt=/dev/stdin
+  COMMAND ${CMAKE_COMMAND} -E
+    remove_directory ${ECLAIR_OUTPUT_DIR}
+  COMMAND ${CMAKE_COMMAND} -E
+    make_directory ${ECLAIR_OUTPUT_DIR}
+  COMMAND ${CMAKE_COMMAND} -E
+    make_directory ${ECLAIR_DATA_DIR}
+  COMMAND ${CMAKE_COMMAND} -E env
+    ECLAIR_DIAGNOSTICS_OUTPUT=${ECLAIR_DIAGNOSTICS_OUTPUT}
+    ECLAIR_DATA_DIR=${ECLAIR_DATA_DIR}
+    CC_ALIASES=${CC_ALIASES}
+    CXX_ALIASES=${CXX_ALIASES}
+    AS_ALIASES=${AS_ALIASES}
+    LD_ALIASES=${LD_ALIASES}
+    AR_ALIASES=${AR_ALIASES}
+    --
+    ${ECLAIR_ENV}
+    -verbose
+    -project_name=${ECLAIR_PROJECT_NAME}
+    -project_root=${ZEPHYR_BASE}
+    -eval_file=${ECLAIR_ECL_DIR}/analysis.ecl
+    -eval_file=${ECLAIR_ECL_DIR}/analysis_${ECLAIR_RULES_SET}.ecl
+    ${ECLAIR_ENV_ADDITIONAL_OPTIONS}
+    -- west build -p always
+                  -b ${BOARD}
+                  --build-dir ${ECLAIR_BUILD_DIR}
+                  ${APPLICATION_SOURCE_DIR}
+  COMMAND ${CMAKE_COMMAND} -E env
+    ECLAIR_DATA_DIR=${ECLAIR_DATA_DIR}
+    ECLAIR_OUTPUT_DIR=${ECLAIR_OUTPUT_DIR}
+    ECLAIR_PROJECT_ECD=${ECLAIR_PROJECT_ECD}
+    --
```

Comment:
The double -- makes the sca build fail with "cmake -E env: unknown option '--'"

Reply:
Hi Simon,
I tried the following command line:
`west build -b mimxrt1064_evk samples/basic/blinky -- -DZEPHYR_SCA_VARIANT=eclair -DECLAIR_CONFIG=$(pwd)/cmake/sca/eclair/.config` and I have no errors.

Can you provide more details?
 -- is present since the first versions, maybe the issue can be related to a different version of cmake?
`simone@beta:~$ cmake --version
cmake version 3.28.2
`

Reply:
I'm using cmake version 3.22.1, that might be the case that the older ones have problems with it.
But we need to keep in mind that zephyrs recommended min version is 3.20.5 so the double -- can cause problems for downstream useres